examples/nlp/token_classification/punctuate_capitalize_infer.py

```diff
@@ -0,0 +1,739 @@
+    punct_logits: torch.Tensor,
+    capit_logits: torch.Tensor,
+    subtokens_mask: torch.Tensor,
+    start_word_ids: Tuple[int],
+    margin: int,
+    is_first: Tuple[bool],
+    is_last: Tuple[bool],
+) -> Tuple[List[np.ndarray], List[np.ndarray], List[int]]:
+    """
+    Applies softmax to get punctuation and capitalization probabilities, applies ``subtokens_mask`` to extract
+    probabilities for words from probabilities for tokens, removes ``margin`` probabilities near edges of a segment.
+    Left margin of the first segment in a query and right margin of the last segment in a query are not removed.
+    Calculates new ``start_word_ids`` taking into the account the margins. If the left margin of a segment is removed
+    corresponding start word index is increased by number of words (number of nonzero values in corresponding
+    ``subtokens_mask``) in the margin.
+    Args:
+        punct_logits: a float tensor of shape ``[batch_size, segment_length, number_of_punctuation_labels]``
+        capit_logits: a float tensor of shape ``[batch_size, segment_length, number_of_capitalization_labels]``
+        subtokens_mask: a float tensor of shape ``[batch_size, segment_length]``
+        start_word_ids: indices of segment first words in a query
+        margin: number of tokens near edges of a segment which probabilities are discarded
+        is_first: is segment the first segment in a query
+        is_last: is segment the last segment in a query
+    Returns:
+        b_punct_probs: list containing ``batch_size`` numpy arrays. The numpy arrays have shapes
+            ``[number_of_word_in_this_segment, number_of_punctuation_labels]``. Word punctuation probabilities for
+            segments in the batch.
+        b_capit_probs: list containing ``batch_size`` numpy arrays. The numpy arrays have shapes
+            ``[number_of_word_in_this_segment, number_of_capitalization_labels]``. Word capitalization probabilities for
+            segments in the batch.
+        new_start_word_ids: indices of segment first words in a query after margin removal
+    """
+    new_start_word_ids = list(start_word_ids)
+    subtokens_mask = subtokens_mask > 0.5
+    b_punct_probs, b_capit_probs = [], []
+    for i, (first, last, pl, cl, stm) in enumerate(zip(is_first, is_last, punct_logits, capit_logits, subtokens_mask)):
+        if not first:
+            new_start_word_ids[i] += torch.count_nonzero(stm[: margin + 1]).numpy()  # + 1 is for [CLS] token
+        stm = _remove_margins(stm, margin, keep_left=first, keep_right=last)
+        for b_probs, logits in [(b_punct_probs, pl), (b_capit_probs, cl)]:
+            p = torch.nn.functional.softmax(
+                _remove_margins(logits, margin, keep_left=first, keep_right=last)[stm], dim=-1,
+            )
+            b_probs.append(p.detach().cpu().numpy())
+    return b_punct_probs, b_capit_probs, new_start_word_ids
+
+
+def add_punctuation_capitalization(
```

Comment:
With this, the model will have two implementations of the same function, I think it's better to keep as it was originally for this PR: extend the functionally of the function and keep only a single version inside the model class